modules/reindex/src/main/java/org/opensearch/index/reindex/remote/RemoteVersion.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.reindex.remote;
+
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents a version of a remote Elasticsearch or OpenSearch cluster for reindexing purposes.
+ * This class provides backward compatibility support for communicating with older Elasticsearch versions
+ * without relying on the global Version class.
+ */
+public final class RemoteVersion implements Comparable<RemoteVersion> {
+
+    private final int major;
+    private final int minor;
+    private final int revision;
+    private final String distribution;
+
+    // Common version constants for backward compatibility
```

Comment:
Maybe it would make sense to name these `ES_<version>` to make it clearer?

Also, I don't think we need all of these versions. I would only keep the ones that are needed for compatibility checks. That is, only keep the versions that are used in code. I noticed references to  `1_0_0`, `2_0_0`, `2_1_0`, `5_0_0`, and `6_3_0`, though I may have missed something. 

Even for the OpenSearch versions, I would only include constants for `1_0_0`, `2_0_0`, and `3_0_0`, but even those aren't strictly necessary, since we haven't changed the search APIs enough to impact reindexing (AFAIK).